test-status/README.md

```diff
@@ -0,0 +1,132 @@
+AMP Test Status Bot
+===================
+
+This GitHub App reports the status of different test types on the AMPHTML
+repository, and allows the build cop to skip flaky tests.
+
+The app runs on an instance of Google AppEngine and is installed exclusively for
+the ampproject/amphtml repository.
+
+
+Interfaces
+----------
+
+### GitHub Webhooks
+
+The App subscribes to the following GitHub Webhooks:
+
+* [`PullRequestEvent`](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+  (`opened` and `synchronize` actions)
+  * Records the head SHA to enable creating checks on the pull request
+
+### API for Travis
+
+The App has the following API points, which are to be triggered from Travis CI
+runs. If the `TRAVIS_IP_ADDRESSES` environment variable is set, only requests
+from this comma separated list of IP addresses will be processed.
+
+* `POST /v0/tests/:headSha/:type/:status(queued|started|skipped)`
+  * Creates a new check on the supplied head commit (for `status` = `queued` or
+    `skipped`) or reports that the `type` tests have started running.
+* `POST /v0/tests/:headSha/:type/report/:passed/:failed`
+  * Updates the equivalent check with the number of `passed` and `failed` tests
+  * If `failed` is 0, sets the check's conclusion to `success`, which turns the
+    check green
+  * If `failed` ≥ 1, sets the check's conclusion to `action_required`, which
+    turns the check red. It also sets a URL to resolve the issue back to the web
+    interface of this app, which can only be accessed by the weekly build cop
```

Comment:
I fear that allowing only the build-cop to resolve a failed run will form a bottleneck. There are likely to be genuine cases where the PR author / reviewer / owner would want to approve a test run. Perhaps we can start with a well-known set of approvers (start with everyone on the infra team) for flaky test runs, and also include the build cop in that list?

Reply:
It's not in the README, but yet - there is also a list of fixed approvers (currently I set it to the infra team, later we can pass the torch to others.) As I mentioned offline, the logic to determine who the build cop is is not there yet :laughing: 

I'll add a mentioned of the fixed list to the README.md though